app/airflow/dags/libs/SR_processing/db_services.py

```diff
@@ -124,6 +124,8 @@ def update_temp_data_dictionary_table(
                     "value",
                     "value_description",
                 ],
+                fast_executemany=True,
+                commit_every=0,  # commit every row to avoid transaction overhead
```

Comment:
It'd be helpful to have this as an environment variable in config, so we tune through configuration rather than deployment.